### PR TITLE
EGM ID Update 80X

### PR DESCRIPTION
**Pull request by lgray**
Forward port of #12738.

New photon ID added to 80X MiniAOD.

### CODE REVIEW

**Comment by lgray** [comment]
@cmsbuild please test

**Comment by lgray** [comment]
@ikrav

File: RecoEgamma/PhotonIdentification/plugins/cuts/PhoAnyPFIsoWithEAAndQuadScalingCut.cc [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    cc.consumes<edm::ValueMap<float> >(contentTags_[anyPFIsoWithEA_]);
+  contentTokens_.emplace(anyPFIsoWithEA_,anyPFIsoWithEA);
+
+  auto rho = cc.consumes<double>(contentTags_[rhoString_]);
+  contentTokens_.emplace(rhoString_, rho);
+}
+
+void PhoAnyPFIsoWithEAAndQuadScalingCut::getEventContent(const edm::EventBase& ev) {  
+  ev.getByLabel(contentTags_[anyPFIsoWithEA_],_anyPFIsoMap);
+  ev.getByLabel(contentTags_[rhoString_],_rhoHandle);
```
Comment (cvuosalo) [inline RecoEgamma/PhotonIdentification/plugins/cuts/PhoAnyPFIsoWithEAAndQuadScalingCut.cc]: Would it be better to use `getByToken()` here?

Comment (lgray) [inline RecoEgamma/PhotonIdentification/plugins/cuts/PhoAnyPFIsoWithEAAndQuadScalingCut.cc]: This is necessary for the classes to operate in (py)FWLite and full CMSSW.
The consumes interface does not exist in ```edm::EventBase```.

Comment (cvuosalo) [inline RecoEgamma/PhotonIdentification/plugins/cuts/PhoAnyPFIsoWithEAAndQuadScalingCut.cc]: An explanatory comment to that effect would be helpful here.

Comment (lgray) [inline RecoEgamma/PhotonIdentification/plugins/cuts/PhoAnyPFIsoWithEAAndQuadScalingCut.cc]: You cannot call consumes in FWLite and thus getbytoken cannot work. Likewise the accessor's are not available in ```edm::EventBase``` and cannot be used in this function.

**Comment by cvuosalo** [comment]
+1

For #12740 28bb2aa449a512aed1caf3d70666d35ef936736c

Adding a new photon ID.

The code changes are satisfactory, and Jenkins [tests](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12740/10229/summary.html) against baseline CMSSW_8_0_X_2015-12-09-1600 show no significant differences. Several other PRs were included by Jenkins in the tests, and they cause differences not related to this PR. An extended test of workflow 134.705_RunMET2015B with 200 events against baseline CMSSW_8_0_X_2015-12-10-1100 also shows no significant differences and no significant change in the size of event content for RECO or Mini-AOD.

**Comment by lgray** [comment]
@monttj @vadler Please review this PR.

File: RecoEgamma/PhotonIdentification/plugins/cuts/PhoAnyPFIsoWithEAAndQuadScalingCut.cc (hunk 1 above)
Comment (davidlange6) [inline RecoEgamma/PhotonIdentification/plugins/cuts/PhoAnyPFIsoWithEAAndQuadScalingCut.cc]: @dr15jones - can we can this behavior?

Comment (Dr15Jones) [inline RecoEgamma/PhotonIdentification/plugins/cuts/PhoAnyPFIsoWithEAAndQuadScalingCut.cc]: @davidlange6 Are you asking if we can get rid of this behavior? That is you want us to invent a new mechanism in FWLite to allow equivalent ```consumes``` behavior?

**Pull request merged**